### PR TITLE
Implementation of configurable timeouts and automatic rollback (v2)

### PR DESCRIPTION
**Pull request by tecracer-theinen**
### Description

**Rewrite of PR #54 for kitchen-vcenter v2**

This patch implements a timeout on waiting for the VM IP address, makes the timeout configurable and adds hints about possible error causes (e.g. VMware tools not being installed, DHCP being unavailable or probable DHCP scope exhaustion).

It also implements an option to automatically remove VMs which failed the connectivity check. These previously cluttered the vCenter service, consumed unnecessary resources and had to be manually cleaned by vCenter administrators from time to time.

Handy additions are outputting the VM name and its IP address for troubleshooting and the use of proper logging for the cloning operation itself.

An additional fix for circumstances where ESX hosts in clusters were not found before is also included.

### Issues Resolved

No related tickets.

### Check List

- [X] All style checks pass.
- [X] Functionality has been documented in the README if applicable
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,6 +116,9 @@ The following parameters should be set in the `driver_config` for the individual
 
 The following parameters should be set in the main `driver_config` section as they are common to all platforms:
  - `vcenter_disable_ssl_verify` - Whether or not to disable SSL verification checks. Good when using self signed certificates. Default: false
+ - `vm_wait_timeout` - How many seconds to wait for VM connectivity. Default: 90
```
Comment (kagarmoe) [inline README.md]:
```suggestion
 - `vm_wait_timeout` - Number of seconds to wait for VM connectivity. Default: 90
```

File: README.md [hunk 2]
```diff
@@ -116,6 +116,9 @@ The following parameters should be set in the `driver_config` for the individual
...
+ - `vm_wait_interval` - Check interval between tries on VM connectivity. Default: 2.0
```
Comment (kagarmoe) [inline README.md]: Is the "Check interval" a length of time? A number of checks?

File: README.md [hunk 3]
```diff
@@ -116,6 +116,9 @@ The following parameters should be set in the `driver_config` for the individual
...
+ - `vm_rollback` - Whether to roll back (destroy) VMs which failed VM connectivity checking. Default: false
```
Comment (kagarmoe) [inline README.md]:
```suggestion
 - `vm_rollback` - Decision to roll back (destroy) VMs failing the VM connectivity check. Default: false
```

File: README.md (hunk 2 above)
Comment (tecracer-theinen) [reply README.md]: It's the length of time between tries to check the VM. So basically "Number of seconds between tries"

File: README.md (hunk 3 above)
Comment (tecracer-theinen) [reply README.md]: I'd propose "Automatic roll back (destroy) of VMs failing the connectivity check. Default: false"? Decision sounds a bit weird to me. Non-native speaker though ;)

**Comment by tecracer-theinen** [comment]
Anyone?

**Comment by tas50** [comment]
Sorry for the delay. I missed this one.

**Pull request merged**